internal/rest/avi_obj_vsvip.go

```diff
@@ -76,44 +91,47 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, vsCach
 		vsvip.VsvipCloudConfigCksum = &cksumstr
 
 		// handling static IP and networkName (infraSetting) updates.
-		vip := &avimodels.Vip{
-			VipID:                  &vipId,
-			AutoAllocateIP:         &autoAllocate,
-			AutoAllocateFloatingIP: vsvip_meta.EnablePublicIP,
-		}
+		if GetIPAMProviderType() != lib.IPAMProviderInfoblox || GetIPAMProviderType() != lib.IPAMProviderCustom {
```

Comment:
A better check is to just do the utils.hasElem check here by declaring a slice variable that lists all of these exceptions in once place.